_pages/dev/ssr/server-side-rendering-optimization.md

```diff
@@ -86,53 +89,109 @@ The full list of parameters is described in the following section.
 
-  **Note**: This property is available in version 3.1.0 and later.
+The default value is `60000` milliseconds (that is, 60 seconds).
 
-- `maxRenderTime` is the maximum amount of time expected for a render to complete. If the render exceeds this timeout, the concurrency slot is released, which allows the next request to be server-side rendered. However, this may not release the rendering resources for a render that has not completed, which may cause additional memory usage on the server. The `maxRenderTime` logs which render exceeds the render time, which is useful for debugging. The value should always be higher than `timeout` and `forcedSsrTimeout`.
+Recommendation: adjust according to your needs.
 
-  The default value is 300,000 milliseconds (5 minutes).
+### maxRenderTime
 
-  **Note**: This property is available in the latest patch versions of 3.1.x and later.
+`maxRenderTime` is the maximum amount of time expected for a render to complete. If the render exceeds this timeout, the concurrency slot is released, which allows the next request to be server-side rendered. However, this may not release the rendering resources for a render that has not completed, which may cause additional memory usage on the server. The `maxRenderTime` logs which render exceeds the render time, which is useful for debugging. The value should always be higher than `timeout` and `forcedSsrTimeout` options. See also [this](#Rendering-of-${URL}-was-not-able-to-complete.-This-might-cause-memory-leaks!)
+
+The default value is `300000` milliseconds (5 minutes).
+
+Recommendation: strongly recommended to set. Adjust according to your needs and expectation.
+
+**Note**: This option is available in the latest patch versions of 3.1.x and later.
+
+### reuseCurrentRendering
+
+`reuseCurrentRendering` - Instead of immediately falling back to CSR while a render for the same rendering key is in progress, this option will make the subsequent requests for this rendering key wait for the current render. All pending requests for the same rendering key will take up only _one_ concurrency slot, because there is only one actual rendering task being performed. Each request independently honors the `timeout` option.
+
+E.g., consider the following setup where `timeout` option is set to 3s, and the given request takes 4s to render. The flow is as follows:
+
+- 1st request arrives and triggers the SSR.
+- 2nd request for the same URL arrives 2s after the 1st one. Instead of falling back to CSR, it waits (with its own timeout) for the render of the first request.
+- 1st request times out after 3s, and falls back to CSR.
+- one second after the timeout, the current render finishes.
+- the 2nd request returns SSR after only 2s of waiting.
+
+Recommendation: recommended to enable, as it will smartly use the serve the server-side renders to multiple requests for the same URL. Might require more server resources (e.g. RAM).
```

Comment:
> it will smartly use the serve the server-side renders to multiple requests for the same URL

Although I know your intention, I'm having hard time to understand this sentence.

Reply:
lol, thanks for catching it :)